pkg/process/metadata/workloadmeta/grpc_test.go

```diff
@@ -216,6 +217,30 @@ func TestStreamServerDropRedundantCacheDiff(t *testing.T) {
 	}, msg)
 }
 
+func TestStreamVersioning(t *testing.T) {
+	extractor, _, conn, stream := setupGRPCTest(t)
+	msg, err := stream.Recv()
+	require.NoError(t, err)
+	assert.EqualValues(t, 0, msg.EventID)
+
+	extractor.diffChan <- &ProcessCacheDiff{cacheVersion: 1} // Simulate a cache update
+	msg, err = stream.Recv()
+	require.NoError(t, err)
+	assert.EqualValues(t, 1, msg.EventID)
+
+	extractor.diffChan <- &ProcessCacheDiff{cacheVersion: 3} // Simulate a missing message
+	_, err = stream.Recv()
+	assert.ErrorContains(t, err, "received version = 3; expected = 2")
+	assert.Equal(t, conn.GetState(), connectivity.Ready) // Assert the underlying connection is still open
+
+	// Make sure we are able to create a new stream using the same connection
+	stream, err = pbgo.NewProcessEntityStreamClient(conn).StreamEntities(context.Background(), &pbgo.ProcessStreamEntitiesRequest{})
+	require.NoError(t, err)
+	msg, err = stream.Recv()
+	assert.EqualValues(t, 0, msg.EventID)
+	require.NoError(t, err)
```

Comment:
This should go before the msg assertion to avoid a npe

Reply:
Good catch